contracts/extensions/interfaces/IMilestonesExtension.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {Metadata} from "../../core/libraries/Metadata.sol";
+
+interface IMilestonesExtension {
+    /// ===============================
+    /// ========== Errors =============
+    /// ===============================
+
+    /// @notice Thrown when the pool manager attempts to lower the max bid
+    error MilestonesExtension_AMOUNT_TOO_LOW();
+
+    /// @notice Thrown when the proposal bid exceeds maximum bid
+    error MilestonesExtension_EXCEEDING_MAX_BID();
+
+    /// @notice Thrown when a recipient is not accepted
+    error MilestonesExtension_INVALID_RECIPIENT();
+
+    /// @notice Thrown when an unauthorized address attempts to submit a milestone
+    error MilestonesExtension_INVALID_SUBMITTER();
+
+    /// @notice Thrown when the milestone is invalid
+    error MilestonesExtension_INVALID_MILESTONE();
+
+    /// @notice Thrown when the new milestone status being reviewed is invalid
+    error MilestonesExtension_INVALID_MILESTONE_STATUS();
+
+    /// @notice Thrown when the milestone are already approved and cannot be changed
+    error MilestonesExtension_MILESTONES_ALREADY_SET();
+
+    /// @notice Thrown when the milestone is not pending
+    error MilestonesExtension_MILESTONE_NOT_PENDING();
+
+    /// @notice Thrown when the milestone is pending
+    error MilestonesExtension_MILESTONE_PENDING();
+
+    /// ===============================
+    /// ========== Events =============
+    /// ===============================
+
+    /// @notice Emitted when the maximum bid is increased.
+    /// @param maxBid The new maximum bid
+    event MaxBidIncreased(uint256 maxBid);
+
+    /// @notice Emitted when a bidder places a bid.
+    /// @param bidderId The address of the bidder
+    /// @param newBid The bid amount
+    event SetBid(address indexed bidderId, uint256 newBid);
+
+    /// @notice Emitted when a milestone is submitted.
+    /// @param milestoneId Id of the milestone
+    event MilestoneSubmitted(uint256 milestoneId);
+
+    /// @notice Emitted for the status change of a milestone.
+    /// @param milestoneId Id of the milestone
+    /// @param status Status of the milestone
+    event MilestoneStatusChanged(uint256 indexed milestoneId, Status status);
+
+    /// @notice Emitted when milestones are set.
+    /// @param milestonesLength Count of milestones
+    event MilestonesSet(uint256 milestonesLength);
+
+    /// ================================
+    /// =========== Enums ==============
+    /// ================================
+
+    /// @notice The Status enum that all milestones are based from
+    enum Status {
```

Comment:
`MilestoneStatus` I think is more clear

Reply:
Absolutely, I actually made that exact change in the RFPSimple branch